src/elasticodbc/drvconn.c

```diff
@@ -207,7 +207,7 @@ BOOL dconn_get_attributes(copyfunc func, const char *connect_string,
     char *last = NULL;
 #endif /* HAVE_STRTOK_R */
 
-    if (our_connect_string = strdup(connect_string), NULL == our_connect_string)
+    if (our_connect_string = strdup(connect_string), free(our_connect_string), NULL == our_connect_string)
```

Comment:
What is the intent here? I don't think this code needs to be changed, in the event that strdup fails (our_connect_string is NULL), it will simply return.

I think this code will force it to always free it after allocation, but not NULL the pointer, so it will free go on to attempt to use unallocated memory. Undefined behaviour when using the our_connect_string after this if statement.

Reply:
Good point; will revert this

Reply:
Reverted